helix-rest/src/main/java/org/apache/helix/rest/metadatastore/MetadataStoreDirectory.java

```diff
@@ -51,6 +52,14 @@
    */
   Collection<String> getAllShardingKeys(String namespace);
 
+  /**
+   * Returns routing data in the given namespace.
+   *
+   * @param namespace namespace in metadata store directory.
+   * @return Routing data map: realm -> List of sharding keys
+   */
+  Map<String, List<String>> getRoutingData(String namespace);
```

Comment:
Nit: should we rename this method to `getAllRoutingDataInNamespace(String namespace)`? That way there's no ambiguity and is more consistent with other methods' names in this class.

Reply:
Changed to `getNamespaceRoutingData(namespace)` to avoid having `of`, `in`.